### PR TITLE
make the Dockerfile build the artifact, not only assemble an image

### PR DESCRIPTION
**Pull request by bystam**
# Jira Issue: [HVG-902]

## What?
Make the `Dockerfile` have the power to build, not only assemble the image.

## Why?
Now we can remove the specific `claims` pipeline and reuse the generic `docker build` one in Codefresh.

## Optional checklist
- [ ] Codescouted
- [ ] Unit tests written
- [x] Tested locally

[HVG-902]: https://hedvig.atlassian.net/browse/HVG-902

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,7 +1,31 @@
...
+RUN mvn clean package
 
-ENTRYPOINT java -jar claims-service-0.0.1-SNAPSHOT.jar
+# Copy test source and build+run tests
+COPY src/test /src/test
+RUN mvn test
+
+
+##### Assemble artifact #####
+FROM amazoncorretto:11-alpine AS assemble
```
Comment (bystam) [inline Dockerfile:18]: I tried using `amazoncorretto` here as well, to not use too many different base images.

File: Dockerfile [hunk 2]
```diff
@@ -1,7 +1,31 @@
...
+
+# Fetch the datadog agent
+RUN apk --no-cache add curl
+RUN curl -o dd-java-agent.jar -L 'https://repository.sonatype.org/service/local/artifact/maven/redirect?r=central-proxy&g=com.datadoghq&a=dd-java-agent&v=LATEST'
```
Comment (bystam) [inline Dockerfile:22]: `alpine` does not have `curl` so first we install that.

File: Dockerfile [hunk 3]
```diff
@@ -1,7 +1,31 @@
...
+
+# Install ffmpeg
+RUN apk --no-cache add ffmpeg
```
Comment (bystam) [inline Dockerfile:25]: `apt` is `apk` in Alpine :)

File: Dockerfile (hunk 1 above)
Comment (bystam) [reply Dockerfile:18]: As in, `openjdk:10.0.2-13-jdk-slim-sid`  which it was previously felt oddly specific.

**Review by meletios — APPROVED** [review]
Sweet! Builds within docker feels amazing. 👏

**Pull request merged**